src/common/contexts/provider.tsx

```diff
@@ -83,105 +81,78 @@ export const ProviderContextProvider: FunctionComponent<ProviderContextProviderP
   networks: supportedChainInfoObjects,
   defaultChainId,
 }) => {
-  const [providerType, setProviderType] = useState<SIGNER_TYPE>(SIGNER_TYPE.IDENTITY);
-  const [isConnected, setIsConnected] = useState<boolean>();
-  const [currentChainId, setCurrentChainId] = useState<ChainId | undefined>(defaultChainId);
-
-  const isSupportedNetwork = (chainId: ChainId) =>
-    supportedChainInfoObjects.some((chainInfoObj) => chainInfoObj.chainId === chainId);
+  const defaultProvider = useRef(createProvider(defaultChainId));
 
-  const defaultProvider = isSupportedNetwork(defaultChainId) ? createProvider(defaultChainId) : undefined;
-  const [providerOrSigner, setProviderOrSigner] = useState<providers.Provider | Signer | undefined>(defaultProvider);
+  const isSupportedNetwork = useCallback(
+    (chainId: ChainId | number | string) =>
+      supportedChainInfoObjects.some((chainInfoObj) => chainInfoObj.chainId.toString() === chainId.toString()),
+    [supportedChainInfoObjects]
+  );
 
-  const updateProviderOrSigner = async (newProviderOrSigner: typeof providerOrSigner) => {
-    try {
-      if (!Signer.isSigner(newProviderOrSigner)) {
-        setIsConnected(false);
-      } else {
-        await (newProviderOrSigner as Signer).getAddress();
-        setIsConnected(true);
-      }
-    } catch (e) {
-      setIsConnected(false);
-    }
-    setProviderOrSigner(newProviderOrSigner);
-  };
+  const [providerType, setProviderType] = useState<SIGNER_TYPE>(SIGNER_TYPE.IDENTITY);
+  const [currentChainId, setCurrentChainId] = useState<ChainId | undefined>(
+    isSupportedNetwork(defaultChainId) ? defaultChainId : undefined
+  );
+  const [account, setAccount] = useState<string | undefined>();
```

Comment:
thoughts: maybe we should move this to [redux](https://github.com/TradeTrust/tradetrust-website/blob/master/src/reducers/sample.ts#L1)?

Reply:
Since the existing setup uses the context already, may we wouldn't want to mix some fields of the provider in the context and some in redux? 

We could, however, change out the entire provider context and move it into a redux state instead if that's preferred. It might be better that way too because the current provider and context setup seem somewhat convoluted in my opinion. But as we have discussed, this will require a major change and is beyond the scope of this refactor. wdyt?

Reply:
i think there are wants by the team to move to redux, but yeah lets stick to this refactor